libraries/botbuilder-dialogs/src/memory/scopes/settingsMemoryScope.ts

```diff
@@ -10,22 +10,23 @@ import { ScopePath } from '../scopePath';
 import { DialogContext } from '../../dialogContext';
 
 /**
- * SettingsMemoryScope maps "settings" -> process.env
+ * SettingsMemoryScope maps "settings" -> dc.context.turnState['settings']
  */
 export class SettingsMemoryScope extends MemoryScope {
     public constructor() {
         super(ScopePath.settings, false);
     }
 
     public getMemory(dc: DialogContext): object {
-        // Clone strings from env
-        const settings: object = {};
+        let settings: object = {};
+        if (dc.context.turnState.has(ScopePath.settings)) {
+            settings = dc.context.turnState.get(ScopePath.settings);
+        }
```

Comment:
This is an additive change in behavior; please add another unit test that demonstrates what happens when the `process.env` key-value pairs collide with the settings from `dc.context.turnState['settings']`.

Reply:
@chon219, is this an attempt to avoid breaking changes for developers getting their settings from `process.env`?

Reply:
@stevengum Yes, I'm trying to make it compatible with previous version, is this a requirement?

Reply:
@chon219, yes this is a requirement.

I think it's safe to assume that the additive code in `SettingsMemoryScope.getMemory()` is what we're going to ship R10 with. Can you add a test for the scenario I described in my [earlier comment](https://github.com/microsoft/botbuilder-js/pull/2217/files#r426188801)?

Reply:
@stevengum sure, I'll add a test later.

Reply:
@stevengum I added a few tests to test if settings from process.env have a higher priority to override settings from turn state.